x-pack/plugins/serverless_search/public/plugin.ts

```diff
@@ -30,10 +31,29 @@ export class ServerlessSearchPlugin
   ): ServerlessSearchPluginSetup {
     core.application.register({
       id: 'serverlessElasticsearch',
-      title: 'Elasticsearch',
+      title: i18n.translate('xpack.serverlessSearch.app.elasticsearch.title', {
+        defaultMessage: 'Elasticsearch',
+      }),
       appRoute: '/app/elasticsearch',
       async mount({ element }: AppMountParameters) {
-        const { renderApp } = await import('./application');
+        const { renderApp } = await import('./application/elasticsearch');
+        const [coreStart, services] = await core.getStartServices();
+        const { security } = services;
+        docLinks.setDocLinks(coreStart.docLinks.links);
+
+        const userProfile = await security.userProfiles.getCurrent();
+
+        return await renderApp(element, coreStart, { userProfile, ...services });
+      },
+    });
+    core.application.register({
+      id: 'serverlessIndexingApi',
```

Comment:
@sphilipse I was torn between registering the indexing api as it's own app with a root URL vs introducing a router to the existing elasticsearch app above here. If you have strong feelings either way let me know.